lib/ret.ex

```diff
@@ -1,158 +1,306 @@
 defmodule Ret do
+  @moduledoc """
+  The boundary of the Ret context.
+  """
+  alias Ret.{Account, Asset, Avatar, AvatarListing, OwnedFile, Project, Repo, Scene, SceneListing, Storage}
   import Canada, only: [can?: 2]
   import Ecto.Query
-  alias Ret.{Account, Asset, Avatar, AvatarListing, OwnedFile, Project, Repo, Scene, SceneListing, Storage}
+  import Ret.Schema, only: [is_serial_id: 1]
 
-  def get_account_by_id(account_id) do
-    Repo.get(Account, account_id)
-  end
+  @doc """
+  Obliterates the account with the given `id` and all its owned assets.
 
-  @asset_file_columns [:asset_owned_file, :thumbnail_owned_file]
-  @project_file_columns [:project_owned_file, :thumbnail_owned_file]
-  @scene_file_columns [:scene_owned_file, :screenshot_owned_file, :model_owned_file]
-  def delete_account(%Account{} = acting_account, %Account{} = account_to_delete) do
-    if can?(acting_account, delete_account(account_to_delete)) do
-      reassign_avatar_listings(account_to_delete, acting_account)
-      reassign_parent_avatars(account_to_delete, acting_account)
-      reassign_scene_listings(account_to_delete, acting_account)
-
-      delete_entities_with_owned_files(
-        from(avatar in Avatar, where: avatar.account_id == ^account_to_delete.account_id),
-        Avatar.file_columns()
-      )
+  Returns `:ok` if successful, otherwise it returns `{:error, reason}`.
 
-      delete_entities_with_owned_files(
-        from(asset in Asset, where: asset.account_id == ^account_to_delete.account_id),
-        @asset_file_columns
-      )
+  Error reasons are:
 
-      project_query = from(project in Project, where: project.created_by_account_id == ^account_to_delete.account_id)
-      scene_query = from(scene in Scene, where: scene.account_id == ^account_to_delete.account_id)
+  * `:not_found` - an account could not be found with the given `id`
+  * `:forbidden` - the user does not have sufficient authorization
+  * `:failed` - the account data was unable to be deleted for an unknown reason
 
-      project_and_scene_owned_files =
-        collect_owned_files(project_query, @project_file_columns) ++
-          collect_owned_files(scene_query, @scene_file_columns)
+  """
+  @spec delete_account(Account.id(), Account.t()) :: :ok | {:error, :not_found | :forbidden | :failed}
+  def delete_account(id, %Account{} = user) when is_serial_id(id) do
+    account_to_delete = Repo.get(Account, id)
 
-      Repo.delete_all(project_query)
-      Repo.delete_all(scene_query)
+    cond do
+      !account_to_delete ->
+        {:error, :not_found}
 
-      delete_owned_files(project_and_scene_owned_files)
+      not can?(user, delete_account(account_to_delete)) ->
+        {:error, :forbidden}
 
-      delete_owned_files(Repo.all(from(o in OwnedFile, where: o.account_id == ^account_to_delete.account_id)))
+      true ->
+        reassignment = [set: [account_id: user.account_id]]
```

Comment:
Trying to understand this syntax: 

Is this an array of tuples `[ {set: [ {account_id: serial_number} ] } ]` OR simply `{Set: {account_id: serial_number }}`